miden/tests/integration/stdlib/math/secp256k1_scalar_field.rs

```diff
@@ -0,0 +1,342 @@
+
+        let d = Self::u288_reduce(&c[5..14], pc);
+        pc = d[8];
+        c[6..14].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[6..15], rhs.limbs[6], &self.limbs);
+
+        let d = Self::u288_reduce(&c[6..15], pc);
+        pc = d[8];
+        c[7..15].copy_from_slice(&d[0..8]);
+
+        Self::u256xu32(&mut c[7..16], rhs.limbs[7], &self.limbs);
+
+        let d = Self::u288_reduce(&c[7..16], pc);
+        pc = d[8];
+        c[8..16].copy_from_slice(&d[0..8]);
+
+        let mut one = Self::one().limbs;
+        for i in 0..8 {
+            one[i] = one[i] * pc;
+        }
+
+        pc = 0;
+
+        (pc, c[8]) = Self::adc(c[8], one[0], pc);
+        (pc, c[9]) = Self::adc(c[9], one[1], pc);
+        (pc, c[10]) = Self::adc(c[10], one[2], pc);
+        (pc, c[11]) = Self::adc(c[11], one[3], pc);
+        (pc, c[12]) = Self::adc(c[12], one[4], pc);
+        (pc, c[13]) = Self::adc(c[13], one[5], pc);
+        (pc, c[14]) = Self::adc(c[14], one[6], pc);
+        (_, c[15]) = Self::adc(c[15], one[7], pc);
+
+        Self::Output {
+            limbs: c[8..16].try_into().expect("incorrect length"),
+        }
+    }
+}
+
+impl PartialEq for ScalarField {
+    /// Checks whether two secp256k1 scalarfield elements are equal or not, in Montogomery form
+    fn eq(&self, other: &Self) -> bool {
+        let mut flg = false;
+
+        for i in 0..8 {
+            flg |= (self.limbs[i] ^ other.limbs[i]) != 0;
+        }
+
+        !flg
+    }
+
+    /// Checks whether two secp256k1 scalar field elements are not equal to each other, in Montogomery form
+    fn ne(&self, other: &Self) -> bool {
+        !(self == other)
+    }
+}
+
+#[test]
+fn test_mul() {
+    let source = "
+    use.std::math::secp256k1_scalar_field
+
+    begin
+        exec.secp256k1_scalar_field::mul
+    end";
+
+    let mut stack = [0u64; 16];
+
+    let mut elm0 = ScalarField::zero();
+    let mut elm1 = ScalarField::zero();
+
+    for i in 0..8 {
+        let a = rand_utils::rand_value::<u32>();
+        let b = rand_utils::rand_value::<u32>();
+
+        stack[i] = a as u64;
+        stack[i ^ 8] = b as u64;
+
+        elm0.limbs[i] = a;
+        elm1.limbs[i] = b;
+    }
+
+    let elm2 = elm0 * elm1;
+    stack.reverse();
+
+    let test = build_test!(source, &stack);
+    let strace = test.get_last_stack_state();
+
+    for i in 0..8 {
+        assert_eq!(Felt::new(elm2.limbs[i] as u64), strace[i]);
+    }
+}
+
+#[test]
+fn test_inv() {
+    let source = "
+    use.std::math::secp256k1_scalar_field
+
+    begin
+        dupw.1
+        dupw.1
+
+        exec.secp256k1_scalar_field::inv
+        exec.secp256k1_scalar_field::mul
+
+        push.801750719
+        assert_eq
+        push.1076732275
+        assert_eq
+        push.1354194884
+        assert_eq
+        push.1162945305
+        assert_eq
+        push.1
+        assert_eq
+        push.0
+        assert_eq
+        push.0
+        assert_eq
+        push.0
+        assert_eq
```

Comment:
nit: I would prefer to check the values resulting from the library calls in the rust test code, e.g. using `expect_stack`, rather than here in the assembly code. What's the reasoning behind doing it this way?

Reply:
Yes, thanks for pointing it out. It should look better in 8d0ae1a12a15d50dced62410f9a4f10ea46ac8b8